MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -2465,63 +2465,64 @@ public void checkBloodnameAdd(Person person, boolean ignoreDice) {
         int bloodnameTarget = 6;
         PersonnelOptions options = person.getOptions();
         Attributes attributes = person.getATOWAttributes();
+        boolean isIlliterate = person.isIlliterate();
         if (!ignoreDice) {
             switch (person.getPhenotype()) {
                 case MEKWARRIOR: {
                     bloodnameTarget += person.hasSkill(SkillType.S_GUN_MEK) ?
                                              person.getSkill(SkillType.S_GUN_MEK)
-                                                   .getFinalSkillValue(options, attributes) :
+                                                   .getFinalSkillValue(options, attributes, isIlliterate) :
```

Comment:
This looks like an absolutely massive amount of changes for one SPA.
Wouldn't it make more sense to store "isIlliterate" state in the ATOWAttributes data, rather than passing it to `getFinalSkillValue()`?
This seems cleaner to me, both for this change, and for any future changes we might need to make to that function.